src/main/drivers/pwm_output.c

```diff
@@ -129,18 +130,73 @@ static void pwmWriteMultiShot(uint8_t index, uint16_t value)
     *motors[index].ccr = lrintf(((float)(value-1000) * MULTISHOT_20US_MULT) + MULTISHOT_5US_PW);
 }
 
-void pwmWriteMotor(uint8_t index, uint16_t value)
+#ifdef USE_DSHOT
+static void pwmWriteDigital(uint8_t index, uint16_t value)
+{
+    pwmMotorOutput_t * const motor = &motors[index];
+
+    motor->value = value;
+    motor->updateFlags |= MOTOR_UPDATE_VALUE;
+}
+
+void pwmCompleteDigitalMotorUpdate(uint8_t motorCount)
+{
+    UNUSED(motorCount);
+
+}
+
+void pwmWriteValueToDmaBuffer(uint16_t value, uint32_t *buffer, uint8_t flags)
+{
+    buffer[0]  = (value & 0x400) ? MOTOR_BIT_1 : MOTOR_BIT_0;
```

Comment:
you may fold it in loop, maybe its easier to read:
```
int csum;
int csum_data = value;
for(int i = 0; i < 3; i++)
  csum ^=  csum_data;   // xor data by nibbles
  csum_data >>= 4;
}
```
when using longer buffer, it would be better to xor data in larger units (char/word) and only finally fold result to 4bits